### PR TITLE
Switch to 4.2.0

### PR DESCRIPTION
**Pull request by dellaert**
Building wheels for 4.2 release

### CODE REVIEW

**Comment by ProfFan** [comment]
Did we cherry-pick all bug-fix PRs to 4.2 branch?

File: build-macos-new.sh [hunk 1]
```diff
@@ -28,8 +28,8 @@ brew install wget "$1" cmake || true
 
 CURRDIR=$(pwd)
 GTSAM_BRANCH="release/4.2"
-GTSAM_LIB_VERSION="4.2.0"
-GTSAM_PYTHON_VERSION="4.2.0"
+GTSAM_LIB_VERSION="4.2"
+GTSAM_PYTHON_VERSION="4.2"
```
Comment (varunagrawal) [inline build-macos-new.sh:32]: This is the Python version *not* the GTSAM version, so it should say 3.x, or better yet read the value from the CI as a passed parameter.

**Comment by varunagrawal** [comment]
@dellaert seems like you forgot to update the version string to remove the "a9" in the 4.2.0 release.

https://github.com/borglab/gtsam/blob/4f66a491ffc83cf092d0d818b11dc35135521612/CMakeLists.txt#L13

**Comment by varunagrawal** [comment]
@dellaert seems like you forgot to update the pre-release version string in the CMake to remove the "a9". It's causing the current failure in the CI.

https://github.com/borglab/gtsam/blob/4f66a491ffc83cf092d0d818b11dc35135521612/CMakeLists.txt#L13

**Comment by varunagrawal** [comment]
I pushed an update to the `varunagrawal/gtsam-manylinux-build` docker image which should allow us to build the python 3.11 wheels.
Once we get that running, I can move the docker image to the Borglab account and set up automated weekly builds so this problem never occurs again.

**Comment by varunagrawal** [comment]
I verified the Python 3.11 build on my fork. Just need the pre-release version bug resolved and we can finish this up!

**Comment by varunagrawal** [comment]
I updated this PR so the python 3.11 wheels should have been built as well if you want to push that.

I put in a hacky fix for the version mismatch so we should be good to go. I can fix it properly after 4.2.1 is released.

**Comment by varunagrawal** [comment]
Going ahead and merging this. We can access the wheels in the Actions tab.

**Pull request merged**